sentinel/src/lib/ids.js

```diff
@@ -64,44 +64,36 @@ const generateId = length => {
   return completedId;
 };
 
-const getIdLengthDoc = db =>
-  new Promise((resolve, reject) =>
-    db.medic.get(ID_LENGTH_DOC_ID, (err, result) => {
-      if (err) {
-        if (err.statusCode !== 404) {
-          return reject(err);
-        } else {
-          result = {
-            _id: ID_LENGTH_DOC_ID,
-          };
-
-          result[ID_LENGTH_PARAM] = INITIAL_ID_LENGTH;
-        }
+const getIdLengthDoc = db => {
+  return db.medic
+    .get(ID_LENGTH_DOC_ID)
+    .catch(err => {
+      if (err.status !== 404) {
+        throw err;
       }
 
-      resolve(result);
-    })
-  );
+      const result = {
+        _id: ID_LENGTH_DOC_ID,
+      };
+
+      result[ID_LENGTH_PARAM] = INITIAL_ID_LENGTH;
+      return result;
+    });
+};
 
-const putIdLengthDoc = (db, idLengthDoc) =>
-  new Promise((resolve, reject) =>
-    db.medic.insert(idLengthDoc, err => {
-      if (err) {
-        // We're OK with a 409, because we're going to presume this is happening
-        // because a human edited it to suite their needs, and their write is more
-        // important than ours.
-        if (err.statusCode === 409) {
-          logger.warn(
-            `409 while trying to store ${idLengthDoc}. If someone edited this document it is expected.`
-          );
-        } else {
-          return reject(err);
-        }
+const putIdLengthDoc = (db, idLengthDoc) => {
+  return db.medic
+    .put(idLengthDoc)
+    .catch(err => {
+      if (err.status !== 409) {
+        throw err;
```

Comment:
I think it's worth keeping the comment about why 409s are ok.

Reply:
Done.